why-choice/index.qmd

```diff
@@ -0,0 +1,265 @@
+---
+title: "Why --choice--"
+description: |
+  As we follow the Data Package standard for structuring metadata, we have to make sure that our metadata files conform to this standard. 
+  This post explains why we chose --choice-- for automating the process of checking metadata files and objects against the standard.
+date: "2024-11-07"
+categories:
+- backend
+- develop
+---
+
+::: content-hidden
+Use other decision posts as inspiration to writing these. Leave the
+content-hidden sections in the text for future reference.
+:::
+
+## Context and problem statement
+
+::: content-hidden
+State the context and some background on the issue, then write a
+statement in the form of a question for the problem.
+:::
+
+We follow the [Data Package](https://datapackage.org/) standard for
+organising metadata. This means that the metadata we keep in JSON files
+have to conform to this standard. These metadata files evolve over the
+life span of a project: they may be generated by Sprout, updated by the
+user through Sprout, and edited directly in a text editor outside of
+Sprout. In order to avoid inconsistencies and prevent errors, every time
+we access or process metadata in Sprout, we have to make sure that they
+still conform to the standard. We therefore need a tool to automate the
+process of checking metadata against the standard. The question then is:
+
+*Which tools are available for this purpose and which one should we
+use?*
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+-   The Data Package standard consists of compulsory requirements and
+    optional recommendations. The former are expressed formally in a
+    [JSON
+    schema](https://datapackage.org/profiles/2.0/datapackage.json),
+    while the latter are presented in a more discursive [textual
+    form](https://datapackage.org/standard/data-package/). An example of
+    an optional recommendation is that the name of packages and
+    resources should follow a particular pattern. Whether we want to
+    adhere to requirements from both of these sources will inform which
+    tool we choose.
+-   We would like the logic for checking metadata against the standard
+    to be extracted from the standard automatically, as far as possible.
+    This way, we wouldn't have to transcribe or translate the
+    requirements of the Data Package standard into Python ourselves,
+    which would be both error prone and subject to falling out of sync
+    with the standard if it's updated.
+-   We already have a set of classes representing the Data Package
+    standard. The aim of these is to allow users to construct metadata
```

Comment:
```suggestion
    standard. Their aim is to allow users to construct metadata
```

Just sounded jagged in my head when I read it like this, I internally wanted to read "these are".